### PR TITLE
Update dependencies and drop Node.js 10 support

### LINKED ISSUE #303: Jasmine 4 support?
Are there plans to support Jasmine 4? karma-jasmine has a hard dependency on jasmine-core ^3.6 so updating jasmine-core locally in my project won't do anything. or is there a way to get karma to use jasmine 4? I am using an Angular app.

### PR DESCRIPTION
**Pull request by XhmikosR**
~~Requires #296 so that we have CI running.~~

Fixes #303, fixes #304 along the way

TODO:

- [x] revert the husky update
- [x] drop Node.js < 12 support
- [x] see if we can update any other deps
- [x] rebase after #313 
- [x] adapt branch protection rules @jginsburgn

### CODE REVIEW

**Comment by nicojs** [comment]
@XhmikosR do you know who has maintainer access here? @johnjbarton isn't responding, would love to get some PR's in.

**Comment by XhmikosR** [comment]
Just be patient, it'll happen when people are back at their offices.

**Comment by nicojs** [comment]
I admire your optimism, but my PR #290 has been waiting for 6 months.

I'm not judging here, I know how it is. Just wondering who to contact 😌

**Comment by XhmikosR** [comment]
Well, I guess @jginsburgn ^^

**Comment by XhmikosR** [comment]
@jginsburgn I'm having trouble with commitlint, I could use a little help. That being said, commitlint should run once in a separate workflow and shouldn't block the other tests.

Also, should I drop Node.js 10 support here? It's causing issues AFAICT

**Comment by sgravrock** [comment]
> Also, should I drop Node.js 10 support here? It's causing issues AFAICT

I'm not sure how much bearing this has on what karma-jasmine does, but FWIW jasmine 4.x requires at least Node 12.17. Older versions can mostly be made to work with a configuration change, at least for now, but we no longer test against them and aren't making any attempt to keep them working.

**Comment by jginsburgn** [comment]
Also, we should probably support Node.js 14-18.

**Comment by devoto13** [comment]
Yeah, I think that bumping the karma version we test against, major Jasmine bump and minimum Node version bump should trigger a major release.

File: package.json
Comment (XhmikosR) [inline package.json:47]: Reminder: we need to revert this so that husky works.

**Comment by jginsburgn** [comment]
Let's merge this PR once https://github.com/karma-runner/karma-jasmine/issues/312 gets settled.

File: package.json
Comment (XhmikosR) [inline package.json:41]: Didn't update this to the latest version due to the `no-var` warnings. I'll leave this to someone else.

**Comment by XhmikosR** [comment]
Rebased and updated to the latest deps.

Personally I would just release this as a major version bump and stop worrying about it :)

**Comment by XhmikosR** [comment]
BTW how am I supposed to construct the commit message so that linting is passing?

**Comment by jginsburgn** [comment]
> Rebased and updated to the latest deps.
> 
> Personally I would just release this as a major version bump and stop worrying about it :)

I agree with you for this release, as latest is already stale. But at least we should couple it with https://github.com/karma-runner/karma-jasmine/pull/304 and all the dependency upgrades we can do! WDYT @devoto13 ?

File: package.json
Comment (jginsburgn) [reply package.json:41]: Can you elaborate on this issue a bit?

> BTW how am I supposed to construct the commit message so that linting is passing?

We are using [Angular's Commit Message Format](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit), which for breaking changes instead of using the exclamation point after the commit type, uses a special kind of footer.

File: package.json
Comment (jginsburgn) [inline package.json:54]: In line with https://github.com/karma-runner/karma-jasmine/pull/304#issuecomment-1096059059, let's replace `*` with `^6.3.16`.

Comment (XhmikosR) [reply package.json]: I went with `^6.0.0` which makes more sense to me (it's a peer dependency and the idea is to allow compatible versions only), but we can change it.

Comment (jginsburgn) [reply package.json]: I have no experience on this. @devoto13 wdyt?

File: .github/workflows/lint.yml
Comment (XhmikosR) [inline .github/workflows/lint.yml:18]: I still think we should use v16 here too. Make sure we are using the latest LTS release like in the release workflow.

File: package.json
Comment (devoto13) [reply package.json]: `^6.0.0` sounds fine to me!

**Review by devoto13 — DISMISSED** [review]
LGTM
I think the open discussions are non-blocking and we can merge this directly.

File: package.json
Comment (XhmikosR) [reply package.json:41]: The newer eslint-config-standard versions have `no-var` set to `warn` so we are flooded with warnings. I suggest someone else deals with this later if/when the supported browsers change.

**Comment by XhmikosR** [comment]
Rebased on last time. I'm still having trouble with the Angular commit messages, so if the commit messages need fixing, feel free to reword :)

Should be ready to land from my side.

**Comment by devoto13** [comment]
@XhmikosR Well, I did just that before your last rebase 😉 

You can set commit message to be the same as in https://github.com/karma-runner/karma-jasmine/commit/1d00309ac0ca7913492d59639f91485c94771857, https://github.com/karma-runner/karma-jasmine/commit/17934c1b224e8505d3675cb4ac66595cf790a60e and https://github.com/karma-runner/karma-jasmine/commit/2aa3e818f8c5f1974a042efa1cc5ca593cd87fc3.

**Comment by XhmikosR** [comment]
@devoto13 thanks, and sorry about that. I missed the force push, I only saw the approval  :/

**Review by devoto13 — APPROVED** [review]
@XhmikosR No worries! 
LGTM

File: package.json
Comment (jginsburgn) [reply package.json:41]: Please excuse me but I still fail to understand. Let's get this conversation resolved to unblock this PR and continue the discussion in https://github.com/karma-runner/karma-jasmine/issues/315.

**Review by jginsburgn — APPROVED** [review]

**Comment by XhmikosR** [comment]
Something else I noticed @jginsburgn @devoto13: should we move jasmine-core to peer dependencies? In major version bump, ofc.

**Comment by jginsburgn** [comment]
I think it makes sense to have it as a dependency! IIUC peer dependencies are meant to be used by plugins to indicate the host package.

**Comment by XhmikosR** [comment]
Yeah, I'm in between myself... I just noticed that other packages use jasmine-core as a peer dependency. Not a big deal since as long as we are up to date package managers can deduplicate the package.

**Comment by devoto13** [comment]
I agree with Jonathan, it's not a plugin and I don't see a good reason why a project must use a single instance of the `jasmine-core`, so I would prefer to keep it as a regular dependency to make `karma-jasmine` easier to install.

**Comment by jginsburgn** [comment]
Then it is settled: let's keep it as a dependency :)

**Pull request merged**
